astroquery/gemini/core.py

```diff
@@ -0,0 +1,480 @@
+            url = "%s/%s" % (url, raw_reduced)
+        response = self._request(method="GET", url=url, data={}, timeout=180, cache=False)
+
+        js = json.loads(response.text)
+        return _gemini_json_to_table(js)
+
+    @class_or_instance
+    def query_raw(self, *args, **kwargs):
+        """
+        perform flexible query against Gemini observations
+
+        This is a more flexible query method.  This method will do special handling for
+        coordinates and radius if present in kwargs.  However, for the remaining arguments
+        it assumes all of args are useable as query path elements.  For kwargs, it assumes
+        all of the elements can be passed as name=value within the query path to Gemini.
+
+        This method does not do any validation checking or attempt to interperet the
+        values being passed, aside from coordinates and radius.
+
+        This method is most useful when the query_criteria and query_region do not
+        meet your needs and you can build the appropriate search in the website.  When
+        you see the URL that is generated by the archive, you can translate that into
+        an equivalent python call with this method.  For example, if the URL in the
+        website is:
+
+        https://archive.gemini.edu/searchform/RAW/cols=CTOWEQ/notengineering/GMOS-N/PIname=Hirst/NotFail
+
+        You can disregard NotFail, cols=x and notengineering.  You would run this query as
+
+        query_raw('GMOS-N', PIname='Hirst')
+
+        Parameters
+        ----------
+        args :
+            The list of parameters to be passed via the query path to the webserver
+        kwargs :
+            The dictionary of parameters to be passed by name=value within the query
+            path to the webserver
+
+        Returns
+        -------
+        response : `~astropy.table.Table`
+        """
+        if 'radius' in kwargs:
+            radius = kwargs['radius']
+        else:
+            radius = None
+        if radius is not None:
+            if isinstance(radius, (int, float)):
+                radius = radius * units.deg
+            radius = astropy.coordinates.Angle(radius)
+        if 'coordinates' in kwargs:
+            coordinates = kwargs['coordinates']
+            coordinates = commons.parse_coordinates(coordinates)
+        else:
+            coordinates = None
+
+        if coordinates is not None:
+            url = "%s/ra=%f/dec=%f" % (url, coordinates.ra.deg, coordinates.dec.deg)
+        if radius is not None:
+            url = "%s/sr=%fd" % (url, radius.deg)
+
+        url = "%s/jsonsummary/notengineering/NotFail" % self.server
+        for arg in args:
+            url = "%s/%s" % (url, arg)
+        for key in filter(lambda key: key not in ["coordinates", "radius"], kwargs):
+            url = "%s/%s=%s" % (url, key, kwargs[key])
+
+        response = self._request(method="GET", url=url, data={}, timeout=180, cache=False)
```

Comment:
same comments as [above](https://github.com/astropy/astroquery/pull/1596/files#diff-15c0230f6c9548d8c5a0a1b2fcdab6b9R316)

Reply:
another thought, I could make separate code to handle url construction for the service to at least move the complexity out of this area and keep it focused

Reply:
Yeah, that's a good idea since you're reusing the URL construction in two places

Reply:
ok, I've come up with something - definitely could use feedback on if that's the direction you were thinking of or if it's just making things too complicated